components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -859,93 +1418,56 @@ private static boolean isDefaultPoliciesExist(Connection connection) throws SQLE
+        try (PreparedStatement statement = connection.prepareStatement(query, new String[]{DAOUtil
+                .getConvertedAutoGeneratedColumnName(dbProductName, "CONDITION_GROUP_ID")})) {
             for (Pipeline pipeline : pipelines) {
                 statement.setString(1, uuid);
                 statement.setString(2, pipeline.getQuotaPolicy().getType());
+                statement.setLong(3, pipeline.getQuotaPolicy().getLimit().getUnitTime());
+                statement.setString(4, pipeline.getQuotaPolicy().getLimit().getTimeUnit());
+                statement.setString(5, pipeline.getDescription());
                 Limit limit = pipeline.getQuotaPolicy().getLimit();
-                if (pipeline.getQuotaPolicy().getType().equals(PolicyConstants.BANDWIDTH_TYPE)) {
-                    if (limit instanceof BandwidthLimit) {
-                        statement.setLong(3, ((BandwidthLimit) limit).getDataAmount());
-                        statement.setString(4, ((BandwidthLimit) limit).getDataUnit());
-                    }
-                } else if (pipeline.getQuotaPolicy().getType().equals(PolicyConstants.REQUEST_COUNT_TYPE)) {
-                    if (limit instanceof RequestCountLimit) {
-                        statement.setLong(3, ((RequestCountLimit) limit).getRequestCount());
-                        statement.setString(4, "");
-                    }
-                }
-                statement.setLong(5, pipeline.getQuotaPolicy().getLimit().getUnitTime());
-                statement.setString(6, pipeline.getQuotaPolicy().getLimit().getTimeUnit());
-                statement.execute();
-                List<Integer> conditionGroupIds = getConditionGroupIDs(connection, uuid);
-                int conID = conditionGroupIds.get(conditionGroupIds.size() - 1); //getting the last condition group id
-                List<Condition> conditionList = pipeline.getConditions();
-                for (Condition condition : conditionList) {
-                    if (condition instanceof IPCondition) {
-                        addIPCondition(connection, (IPCondition) condition, conID);
-                    } else if (condition instanceof HeaderCondition) {
-                        addHeaderCondition(connection, (HeaderCondition) condition, conID);
-                    } else if (condition instanceof JWTClaimsCondition) {
-                        addJWTClaimCondition(connection, (JWTClaimsCondition) condition, conID);
-                    } else if (condition instanceof QueryParameterCondition) {
-                        addParamCondition(connection, (QueryParameterCondition) condition, conID);
+                setDefaultThrottlePolicyDetailsPreparedStmt(limit, statement);
+                statement.executeUpdate();
+                ResultSet rs = statement.getGeneratedKeys();
+                if (rs.next()) {
+                    // get the auto increment id
+                    int conditionId = rs.getInt(1);
```

Comment:
The column name for the auto generated id has been passed to the prepared statement. Wouldn't this do?